src/components/MovieDetails/MovieDetails.js

```diff
@@ -1,26 +1,43 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import { useParams, Link } from 'react-router-dom';
+import { getMoviesFromIdThunk } from '../../store/middlewares';
+
 import styles from './MovieDetails.css';
 
-const MovieDetails = ({ movieData, triggerMovieDetails }) => {
+const MovieDetails = (props) => {
+  const { id } = useParams();
+  props.dispatch(getMoviesFromIdThunk(id));
+
+  const { movie } = props;
+
   return (
-    <div className={styles.wrapper}>
-      <div className={styles.header}>
-        <span className={styles.closeIcon} onClick={triggerMovieDetails} />
-      </div>
-      <div className={styles.movieWrapper}>
-        <img className={styles.posterPath} src={movieData.poster_path} />
-        <div className={styles.movieInfo}>
-          <div className={styles.movieHeader}>
-            <h2 className={styles.title}>{movieData.title}</h2>
-            <p className={styles.vote}>{movieData.vote_average}</p>
+    <Link to="/">
+      {movie ? (
+        <div className={styles.wrapper}>
+          <div className={styles.header}>
+            <span className={styles.closeIcon} />
           </div>
-          <div>
-            <p>{movieData.overview}</p>
+          <div className={styles.movieWrapper}>
+            <img className={styles.posterPath} src={movie.poster_path} />
+            <div className={styles.movieInfo}>
+              <div className={styles.movieHeader}>
+                <h2 className={styles.title}>{movie.title}</h2>
+                <p className={styles.vote}>{movie.vote_average}</p>
+              </div>
+              <div>
+                <p>{movie.overview}</p>
+              </div>
+            </div>
           </div>
         </div>
-      </div>
-    </div>
+      ) : null}
+    </Link>
   );
 };
 
-export default MovieDetails;
+const mapStateToProps = (state) => ({
```

Comment:
instead of map state to props, you could use `import { useSelector } from 'react-redux';`